subsys/bluetooth/host/CMakeLists.txt

```diff
@@ -126,3 +126,9 @@ if(CONFIG_BT_CONN_DISABLE_SECURITY)
   Do not use in production."
     )
 endif()
+if((NOT CONFIG_BT_RECV_IS_RX_THREAD) AND
+   (CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE LESS 2200))
+  message(WARNING "For Bluetooth Settings function.
+  It is recommended that the system work thread stack space is 2200 and above."
+    )
+endif()
```

Comment:
Seems to be missing a check for CONFIG_BT_SETTINGS, but otherwise this is a decent workaround if we can't add the dependency for `CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE`

Reply:
Good eyesight.